sdk/core/azure-core-amqp/src/amqp/private/management_impl.hpp

```diff
@@ -93,8 +129,8 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
         Models::AmqpMessage>
         m_messageQueue;
 
+#if UAMQP_MANAGEMENT_IMPLEMENTATION
```

Comment:
should this condition across many places remain `#if UAMQP_MANAGEMENT_IMPLEMENTATION`, or should it be `#if defined(UAMQP_MANAGEMENT_IMPLEMENTATION)`?

Reply:
> <b>#if, #elif </b>
> The expression is a constant expression, using only [constants](https://en.cppreference.com/w/c/language/expressions#Constants_and_literals) and identifiers, defined using [#define](https://en.cppreference.com/w/c/preprocessor/replace) directive. Any identifier, which is not literal, non defined using [#define](https://en.cppreference.com/w/c/preprocessor/replace) directive, evaluates to 0.

so using #if works if it's been defined or not, while #if defined means that if you say `-DUAMQP_MANAGEMENT_IMPLEMENTATION=1` on the command line, the conditional won't work.

I don't feel strongly about this, but because `#if` is a superset of `#if defined(`, I tend to prefer `#if <foo>` over `#if defined(foo)`.